provider/oci/images.go

```diff
@@ -0,0 +1,539 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package oci
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/juju/errors"
+	// jujuos "github.com/juju/utils/os"
+	"github.com/juju/utils/series"
+
+	"github.com/juju/juju/environs/imagemetadata"
+	"github.com/juju/juju/environs/instances"
+	"github.com/juju/juju/provider/oci/common"
+
+	ociCore "github.com/oracle/oci-go-sdk/core"
+)
+
+const (
+	BareMetal      InstanceType = "metal"
+	VirtualMachine InstanceType = "vm"
+	GPUMachine     InstanceType = "gpu"
+
+	// ImageTypeVM should be run on a virtual instance
+	ImageTypeVM ImageType = "vm"
+	// ImageTypeBM should be run on bare metal
+	ImageTypeBM ImageType = "metal"
+	// ImageTypeGPU should be run on an instance with attached GPUs
+	ImageTypeGPU ImageType = "gpu"
+	// ImageTypeGeneric should work on any type of instance (bare metal or virtual)
+	ImageTypeGeneric ImageType = "generic"
+
+	windowsOS = "Windows"
+	centOS    = "CentOS"
+	ubuntuOS  = "Canonical Ubuntu"
+)
+
+// shapeSpecs is a map containing resource information
+// about each shape. Unfortunately the API simply returns
+// the name of the shape and nothing else. For details see:
+// https://cloud.oracle.com/infrastructure/pricing
+// https://cloud.oracle.com/infrastructure/compute/pricing
+var shapeSpecs = map[string]ShapeSpec{
```

Comment:
can we move this to a shapespecs.go file
since it is just a bunch of constants, which we may want to autogenerate.
I realize it won't be autogenerated in any short term, but at least it pulls out this from being a distraction while trying to understand the logic in this file.